### PR TITLE
ci: split headless-ssr in dev and prod

### PR DESCRIPTION
**Pull request by alexprudhomme**
https://coveord.atlassian.net/browse/KIT-2783

### CODE REVIEW

File: .github/workflows/prbot.yml [hunk 1]
```diff
@@ -189,14 +189,22 @@ jobs:
...
+    name: 'Run e2e tests on Headless SSR sample in prod'
     needs: build
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: ./.github/actions/setup
-      - uses: ./.github/actions/e2e-headless-ssr
+      - uses: ./.github/actions/e2e-headless-ssr-dev
+  e2e-headless-ssr-test-prod:
+    name: 'Run e2e tests on Headless SSR sample in dev'
```
Comment (mrrajamanickam-coveo) [inline .github/workflows/prbot.yml:201]:
```suggestion
    name: 'Run e2e tests on Headless SSR sample in prod'
```

Comment (mrrajamanickam-coveo) [reply .github/workflows/prbot.yml:201]: similar name mismatches above as well

**Review by mrrajamanickam-coveo — CHANGES_REQUESTED** [review]

File: .github/workflows/prbot.yml (hunk 1 above)
Comment (alexprudhomme) [reply .github/workflows/prbot.yml:201]: oops !

File: .github/workflows/masterbot.yml [hunk 2]
```diff
@@ -116,14 +138,22 @@ jobs:
       - uses: actions/checkout@v4
       - uses: ./.github/actions/setup
       - uses: ./.github/actions/e2e-atomic-insight-panel
-  e2e-headless-ssr-test:
-    name: 'Run e2e tests on Headless SSR sample'
+  e2e-headless-ssr-test-dev:
+    name: 'Run e2e tests on Headless SSR sample in prod'
```
Comment (mrrajamanickam-coveo) [inline .github/workflows/masterbot.yml:142]: misspell here as well and below

**Review by mrrajamanickam-coveo — APPROVED** [review]

**Review by dmbrooke — APPROVED** [review]

**Pull request merged**